### PR TITLE
Load scheduler state when resuming training

### PR DESCRIPTION
**Pull request by n-poulsen**
This pull request addresses learning rate schedulers when resuming training. Currently, the schedulers are simply built from the config file. So when continuing to train from an existing snapshot, the scheduler will restart at epoch 0, and the learning rate won't be adapted (as mentioned in #2784).

In this pull request, the code is updated to:

- save the scheduler state dicts in snapshots
- when resuming training, try to load the state dict for the scheduler
  - if successful, set the optimizers learning rate to the last learning rate from the scheduler
- a `load_scheduler_state_dict` key is added to the runner configuration
  - the default is `True`: loading a snapshot with a saved scheduler to continue training will load the scheduler state dict
  - however, users might edit the scheduler's parameters and want an updated learning rate to continue training
  - in that case, they need to set `load_scheduler_state_dict: false` so the state dict doesn't overwrite their edited parameters
  - the `self.starting_epoch` value is used to set the `last_epoch` in the scheduler; so the way the learning rate will be schedule matches the epochs printed in the logs

Tests were added to validate that the expected learning rates are applied.

### CODE REVIEW

File: deeplabcut/pose_estimation_pytorch/runners/train.py [hunk 1]
```diff
@@ -89,27 +99,34 @@ def __init__(
         self._print_valid_loss = True
 
         if self.snapshot_path is not None and self.snapshot_path != "":
```
Comment (maximpavliv) [inline deeplabcut/pose_estimation_pytorch/runners/train.py]:
```suggestion
        if self.snapshot_path:
```

File: deeplabcut/pose_estimation_pytorch/runners/train.py [hunk 2]
```diff
@@ -45,33 +45,43 @@ class TrainingRunner(Runner, Generic[ModelType], metaclass=ABCMeta):
     def __init__(
```
Comment (maximpavliv) [inline deeplabcut/pose_estimation_pytorch/runners/train.py:68]: I think the docstring of the class is wrong (it is the same as the base Runner's one)

The InferenceRunner's docstring should probably be updated as well

**Review by maximpavliv — APPROVED** [review]
